File: viralt-app/src/types/winner.ts
/** Winner Types -- Types for the winner selection system */

/** Main winner information */
export interface WinnerInfo {
  winnerId: number;
  campaignId: number;
  clientId: number;
  prizeId: number | null;
  selectedAt: string;
  selectionMethod: number;
  notified: boolean;
  claimed: boolean;
  claimData: string | null;
}

/** Winner list API result */
export interface WinnerListResult {
  sucesso: boolean;
  mensagem: string;
  erros: string[] | null;
  winners: WinnerInfo[];
}

/** Draw request payload */
export interface DrawRequest {
  winnerCount: number;
  selectionMethod: number;
}

/** Status-only result */
export interface StatusResult {
  sucesso: boolean;
  mensagem: string;
  erros: string[] | null;
}

File: Viralt.Domain/Models/Winner.cs
namespace Viralt.Domain.Models;

public class Winner
{
    public long WinnerId { get; set; }
    public long CampaignId { get; set; }
    public long ClientId { get; set; }
    public long? PrizeId { get; set; }
    public DateTime SelectedAt { get; set; }
    public int SelectionMethod { get; set; }
    public bool Notified { get; set; }
    public bool Claimed { get; set; }
    public string ClaimData { get; set; }

    public virtual Campaign Campaign { get; set; }
    public virtual Client Client { get; set; }
    public virtual Prize Prize { get; set; }
}
